pkg/machines/reducers.es6

```diff
@@ -184,8 +186,27 @@ function ui(state, action) {
         case 'VM_INSTALL_COMPLETED':
             delete state.vmsInstallInitiated[action.vm.name];
             return state;
+        case 'ADD_NOTIFICATION': {
+            const notification = typeof action.notification === 'string' ? { message: action.notification } : action.notification;
+            const notifs = state.notifications;
+            notification.id = notifs.length > 0 ? notifs[notifs.length - 1].id + 1 : 1;
```

Comment:
This doesn't work. Once you clear some (but not all) notifications, the next one will run into ID collisions. I suggest to use a global  counter instead.

Reply:
why it shouldn't work? The highest id is always at the end, even if you delete some ids in the middle.

This reuses id space if you delete ids from the end and you can always expect the same ids (e.g. in tests). This is not true with global counter (unless you reset it which has the same problems as follows).

The only collision happens if you have old information about the store state (id is not meant to be internal) and try to delete  something. Then it would collide.

Reply:
Right; sorry, I misread that.